--- src/main/cpp/socketoutputstream.cpp.old	Thu Jul 10 00:41:41 2008
+++ src/main/cpp/socketoutputstream.cpp	Thu Jul 10 00:41:47 2008
@@ -49,7 +49,7 @@
   if (buf.remaining() > 0) {
     size_t sz = array.size();
     array.resize(sz + buf.remaining());
-    memcpy(&array[sz], buf.current(), buf.remaining());
+    std::memcpy(&array[sz], buf.current(), buf.remaining());
     buf.position(buf.limit());
   }
 }
