--- ./src/SConscript.old	Wed Jul 16 15:30:52 2008
+++ ./src/SConscript	Wed Jul 16 15:36:03 2008
@@ -35,8 +35,8 @@
 if env['PLATFORM'] == 'Darwin':
     libraries.append('iconv')
     linkflags = '-Wl,-install_name,libmapnik.dylib'
-elif env['PLATFORM'] == 'SunOS' and env['CXX'].startswith('CC'):
-    linkflags = '-R. -h libmapnik.so'
+elif env['PLATFORM'] == 'SunOS':
+    linkflags = '-R/opt/boost/lib -R/opt/ts/lib -h libmapnik.so'
 else: # Linux and others
     linkflags = '-Wl,-rpath-link,. -Wl,-soname,libmapnik.so.' + ("%d.%d" % (ABI_VERSION[0],ABI_VERSION[1])) 
 
