--- scripts/mysqld_safe.sh.old	Tue Mar  3 03:22:53 2009
+++ scripts/mysqld_safe.sh	Tue Mar  3 03:24:45 2009
@@ -542,6 +542,10 @@
 # Avoid 'nohup: ignoring input' warning
 test -n "$NOHUP_NICENESS" && cmd="$cmd < /dev/null"
 
+LD_PRELOAD_32=/opt/ts/lib/libfasttime.so
+LD_PRELOAD_64=/opt/ts/lib/amd64/libfasttime.so
+export LD_PRELOAD_32 LD_PRELOAD_64
+
 log_notice "Starting $MYSQLD daemon with databases from $DATADIR"
 while true
 do
