--- mytop.old	Tue Mar  3 04:50:15 2009
+++ mytop	Sun Mar  1 19:28:44 2009
@@ -803,8 +803,8 @@
 
         ## Queries per second...
 
-        my $avg_queries_per_sec  = sprintf("%.2f", $STATUS{Questions} / $STATUS{Uptime});
-        my $num_queries          = $STATUS{Questions};
+        my $avg_queries_per_sec  = sprintf("%.2f", $STATUS{Queries} / $STATUS{Uptime});
+        my $num_queries          = $STATUS{Queries};
 
         my @t = localtime(time);
 
@@ -823,25 +823,25 @@
 
 
         printf " Queries: %-5s  qps: %4.0f Slow: %7s         Se/In/Up/De(%%):    %02.0f/%02.0f/%02.0f/%02.0f \n",
-               make_short( $STATUS{Questions} ),  # q total
-               $STATUS{Questions} / $STATUS{Uptime},  # qps, average
+               make_short( $STATUS{Queries} ),  # q total
+               $STATUS{Queries} / $STATUS{Uptime},  # qps, average
                make_short( $STATUS{Slow_queries} ),    # slow
 
                # hmm. a Qcache hit is really a select and should be counted.
-               100 * ($STATUS{Com_select} + ($STATUS{Qcache_hits}||0) )    / $STATUS{Questions},
-               100 * ($STATUS{Com_insert} +  $STATUS{Com_replace} ) / $STATUS{Questions},
-               100 * ($STATUS{Com_update} )  / $STATUS{Questions},
-               100 * $STATUS{Com_delete} / $STATUS{Questions};
+               100 * ($STATUS{Com_select} + ($STATUS{Qcache_hits}||0) )    / $STATUS{Queries},
+               100 * ($STATUS{Com_insert} +  $STATUS{Com_replace} ) / $STATUS{Queries},
+               100 * ($STATUS{Com_update} )  / $STATUS{Queries},
+               100 * $STATUS{Com_delete} / $STATUS{Queries};
 
         $lines_left--;
 
         if ($t_delta)
         {
-          my $q_diff = ( $STATUS{Questions} - $OLD_STATUS{Questions} );
+          my $q_diff = ( $STATUS{Queries} - $OLD_STATUS{Queries} );
 #          print("q_diff: $STATUS{Questions} - $OLD_STATUS{Questions}  / $t_delta = $q_diff\n");
 
           printf("             qps now: %4.0f Slow qps: %3.1f  Threads: %4.0f (%4.0f/%4.0f) %02.0f/%02.0f/%02.0f/%02.0f \n",
-                 ( $STATUS{Questions} - $OLD_STATUS{Questions} ) / $t_delta,
+                 ( $STATUS{Queries} - $OLD_STATUS{Queries} ) / $t_delta,
                  ( # slow now (qps)
                   ($STATUS{Slow_queries} ) ?
                   ( $STATUS{Slow_queries} - $OLD_STATUS{Slow_queries} ) / $t_delta :
@@ -1068,7 +1068,7 @@
 
 sub GetQPS()
 {
-    my($data) = Hashes('SHOW STATUS LIKE "Questions"');
+    my($data) = Hashes('SHOW STATUS LIKE "Queries"');
     my $num   = $data->{Value};
 
     if (not defined $questions) ## first time?
