--- src/nrpe.c.orig	Tue Jan 13 07:25:51 2009
+++ src/nrpe.c	Tue Jan 13 07:25:54 2009
@@ -613,10 +613,14 @@
 		log_facility=LOG_UUCP;
 	else if(!strcmp(varvalue,"cron"))
 		log_facility=LOG_CRON;
+#ifdef LOG_AUTHPRIV
 	else if(!strcmp(varvalue,"authpriv"))
 		log_facility=LOG_AUTHPRIV;
+#endif
+#ifdef LOG_FTP
 	else if(!strcmp(varvalue,"ftp"))
 		log_facility=LOG_FTP;
+#endif
 	else if(!strcmp(varvalue,"local0"))
 		log_facility=LOG_LOCAL0;
 	else if(!strcmp(varvalue,"local1"))
