--- Makefile.in.old	Mon Feb 23 23:31:11 2009
+++ Makefile.in	Mon Feb 23 23:31:40 2009
@@ -84,22 +84,22 @@
 test/test_ssl.lo: test/test_ssl.c $(HEADERS)
 
 $(TARGET_LIB): $(OBJECTS)
-	$(LIBTOOL) $(LTFLAGS) --mode=link $(CC) $(LDFLAGS) -rpath $(libdir) -o $@ $^ $(LIBS)
+	$(LIBTOOL) $(LTFLAGS) --mode=link $(CC) $(CFLAGS) $(LDFLAGS) -rpath $(libdir) -o $@ $^ $(LIBS)
 
 test/serf_get: $(TARGET_LIB) test/serf_get.lo
-	$(LIBTOOL) $(LTFLAGS) --mode=link $(CC) $(LDFLAGS) -static -o $@ $^ $(LIBS)
+	$(LIBTOOL) $(LTFLAGS) --mode=link $(CC) $(CFLAGS) $(LDFLAGS) -static -o $@ $^ $(LIBS)
 
 test/serf_response: $(TARGET_LIB) test/serf_response.lo
-	$(LIBTOOL) $(LTFLAGS) --mode=link $(CC) $(LDFLAGS) -static -o $@ $^ $(LIBS)
+	$(LIBTOOL) $(LTFLAGS) --mode=link $(CC) $(CFLAGS) $(LDFLAGS) -static -o $@ $^ $(LIBS)
 
 test/serf_request: $(TARGET_LIB) test/serf_request.lo
-	$(LIBTOOL) $(LTFLAGS) --mode=link $(CC) $(LDFLAGS) -static -o $@ $^ $(LIBS)
+	$(LIBTOOL) $(LTFLAGS) --mode=link $(CC) $(CFLAGS) $(LDFLAGS) -static -o $@ $^ $(LIBS)
 
 test/serf_spider: $(TARGET_LIB) test/serf_spider.lo
-	$(LIBTOOL) $(LTFLAGS) --mode=link $(CC) $(LDFLAGS) -static -o $@ $^ $(LIBS)
+	$(LIBTOOL) $(LTFLAGS) --mode=link $(CC) $(CFLAGS) $(LDFLAGS) -static -o $@ $^ $(LIBS)
 
 test/test_all: $(TARGET_LIB) $(TEST_SUITE_OBJECTS)
-	$(LIBTOOL) $(LTFLAGS) --mode=link $(CC) $(LDFLAGS) -static -o $@ $^ $(LIBS)
+	$(LIBTOOL) $(LTFLAGS) --mode=link $(CC) $(CFLAGS) $(LDFLAGS) -static -o $@ $^ $(LIBS)
 
 check: test/serf_response test/test_all
 	@for i in $(TESTCASES); \
