--- configure.in.old	Wed Mar 18 07:13:07 2009
+++ configure.in	Wed Mar 18 07:13:46 2009
@@ -788,8 +788,10 @@
 
 dnl Check if we want to compile everything with static
 AC_ARG_ENABLE(static,[  --enable-static         Build statically linked binaries],
-[ LDFLAGS="${LDFLAGS} -static";
- enable_static=yes])
+[case "$enableval" in
+  yes) LDFLAGS="${LDFLAGS} -static"; enable_static=yes;;
+  *) ;;
+esac])
 
 AC_ARG_ENABLE(server,[  --enable-server         Turn on build of server],
 [case "${enableval}" in
