Index: includes/Namespace.php
===================================================================
--- includes/Namespace.php	(revision 31817)
+++ includes/Namespace.php	(revision 31818)
@@ -175,7 +175,9 @@
 	 * @return bool
 	 */
 	function isMovable() {
-		return $this->isMovable;	
+		#return $this->isMovable;	
+		# FIXME: This means that classes with custom handlers are assumed to never be movable
+		return $this->isMovable && empty($this->handlerClass);
 	}
 
 	/**
