Index: imagemagick-6.3.7.9.dfsg1/utilities/animate.1.in
===================================================================
--- imagemagick-6.3.7.9.dfsg1.orig/utilities/animate.1.in	2007-12-26 04:10:09.000000000 +0100
+++ imagemagick-6.3.7.9.dfsg1/utilities/animate.1.in	2008-01-10 20:45:49.000000000 +0100
@@ -86,7 +86,7 @@
 
 Buttons: 
   1    press to map or unmap the Command widget
-.SH SEE-ALSO
+.SH SEE ALSO
 ImageMagick(1)
 
 .SH COPYRIGHT
Index: imagemagick-6.3.7.9.dfsg1/utilities/compare.1.in
===================================================================
--- imagemagick-6.3.7.9.dfsg1.orig/utilities/compare.1.in	2007-12-26 04:10:09.000000000 +0100
+++ imagemagick-6.3.7.9.dfsg1/utilities/compare.1.in	2008-01-10 20:45:49.000000000 +0100
@@ -51,7 +51,7 @@
   -version             print version information
 
 By default, the image format of `file' is determined by its magic number.  To specify a particular image format, precede the filename with an image format name and a colon (i.e. ps:image) or specify the image type as the filename suffix (i.e. image.ps).  Specify 'file' as '-' for standard input or output.
-.SH SEE-ALSO
+.SH SEE ALSO
 ImageMagick(1)
 
 .SH COPYRIGHT
Index: imagemagick-6.3.7.9.dfsg1/utilities/composite.1.in
===================================================================
--- imagemagick-6.3.7.9.dfsg1.orig/utilities/composite.1.in	2007-12-26 04:10:09.000000000 +0100
+++ imagemagick-6.3.7.9.dfsg1/utilities/composite.1.in	2008-01-10 20:45:49.000000000 +0100
@@ -93,7 +93,7 @@
   -version             print version information
 
 By default, the image format of `file' is determined by its magic number.  To specify a particular image format, precede the filename with an image format name and a colon (i.e. ps:image) or specify the image type as the filename suffix (i.e. image.ps).  Specify 'file' as '-' for standard input or output.
-.SH SEE-ALSO
+.SH SEE ALSO
 ImageMagick(1)
 
 .SH COPYRIGHT
Index: imagemagick-6.3.7.9.dfsg1/utilities/conjure.1.in
===================================================================
--- imagemagick-6.3.7.9.dfsg1.orig/utilities/conjure.1.in	2007-12-26 04:10:09.000000000 +0100
+++ imagemagick-6.3.7.9.dfsg1/utilities/conjure.1.in	2008-01-10 20:45:49.000000000 +0100
@@ -26,7 +26,7 @@
 In additiion, define any key value pairs required by your script.  For example,
 
     conjure -size 100x100 -color blue -foo bar script.msl
-.SH SEE-ALSO
+.SH SEE ALSO
 ImageMagick(1)
 
 .SH COPYRIGHT
Index: imagemagick-6.3.7.9.dfsg1/utilities/convert.1.in
===================================================================
--- imagemagick-6.3.7.9.dfsg1.orig/utilities/convert.1.in	2007-12-26 04:10:09.000000000 +0100
+++ imagemagick-6.3.7.9.dfsg1/utilities/convert.1.in	2008-01-10 20:45:49.000000000 +0100
@@ -238,7 +238,7 @@
   -version             print version information
 
 By default, the image format of `file' is determined by its magic number.  To specify a particular image format, precede the filename with an image format name and a colon (i.e. ps:image) or specify the image type as the filename suffix (i.e. image.ps).  Specify 'file' as '-' for standard input or output.
-.SH SEE-ALSO
+.SH SEE ALSO
 ImageMagick(1)
 
 .SH COPYRIGHT
Index: imagemagick-6.3.7.9.dfsg1/utilities/display.1.in
===================================================================
--- imagemagick-6.3.7.9.dfsg1.orig/utilities/display.1.in	2007-12-26 04:10:09.000000000 +0100
+++ imagemagick-6.3.7.9.dfsg1/utilities/display.1.in	2008-01-10 20:45:49.000000000 +0100
@@ -116,7 +116,7 @@
   1    press to map or unmap the Command widget
   2    press and drag to magnify a region of an image
   3    press to load an image from a visual image directory
-.SH SEE-ALSO
+.SH SEE ALSO
 ImageMagick(1)
 
 .SH COPYRIGHT
Index: imagemagick-6.3.7.9.dfsg1/utilities/identify.1.in
===================================================================
--- imagemagick-6.3.7.9.dfsg1.orig/utilities/identify.1.in	2007-12-26 04:10:09.000000000 +0100
+++ imagemagick-6.3.7.9.dfsg1/utilities/identify.1.in	2008-01-10 20:45:49.000000000 +0100
@@ -50,7 +50,7 @@
   -version             print version information
 
 By default, the image format of `file' is determined by its magic number.  To specify a particular image format, precede the filename with an image format name and a colon (i.e. ps:image) or specify the image type as the filename suffix (i.e. image.ps).  Specify 'file' as '-' for standard input or output.
-.SH SEE-ALSO
+.SH SEE ALSO
 ImageMagick(1)
 
 .SH COPYRIGHT
Index: imagemagick-6.3.7.9.dfsg1/utilities/ImageMagick.1.in
===================================================================
--- imagemagick-6.3.7.9.dfsg1.orig/utilities/ImageMagick.1.in	2007-12-05 23:22:41.000000000 +0100
+++ imagemagick-6.3.7.9.dfsg1/utilities/ImageMagick.1.in	2008-01-10 20:45:49.000000000 +0100
@@ -5,7 +5,7 @@
 \fBconvert\fP \fIinput-file\fP [\fIoptions\fP] \fIoutput-file\fP
 .SH OVERVIEW
 
-ImageMagick®, is a software suite to create, edit, and compose bitmap images. It can read, convert and write images in a variety of formats (about 100) including GIF, JPEG, JPEG-2000, PNG, PDF, PhotoCD, TIFF, and DPX. Use ImageMagick to translate, flip, mirror, rotate, scale, shear and transform images, adjust image colors, apply various special effects, or draw text, lines, polygons, ellipses and B\\['e]zier curves.
+ImageMagick\[rg], is a software suite to create, edit, and compose bitmap images. It can read, convert and write images in a variety of formats (about 100) including GIF, JPEG, JPEG-2000, PNG, PDF, PhotoCD, TIFF, and DPX. Use ImageMagick to translate, flip, mirror, rotate, scale, shear and transform images, adjust image colors, apply various special effects, or draw text, lines, polygons, ellipses and B\\['e]zier curves.
 
 ImageMagick is free software delivered as a ready-to-run binary distribution or as source code that you can freely use, copy, modify, and distribute. Its license is compatible with the GPL. It runs on all major operating systems.
 
@@ -62,7 +62,7 @@
 interprets and executes scripts written in the Magick Scripting Language (MSL).
 .PP
 For more information about the ImageMagick, point your browser to file://@DOCUMENTATION_PATH@/index.html or http://www.imagemagick.org/.
-.SH SEE-ALSO
+.SH SEE ALSO
 convert(1), identify(1), composite(1), montage(1), compare(1), display(1), animate(1), import(1), conjure(1), quantize(5), miff(4)
 
 .SH COPYRIGHT
Index: imagemagick-6.3.7.9.dfsg1/utilities/import.1.in
===================================================================
--- imagemagick-6.3.7.9.dfsg1.orig/utilities/import.1.in	2007-12-26 04:10:09.000000000 +0100
+++ imagemagick-6.3.7.9.dfsg1/utilities/import.1.in	2008-01-10 20:45:49.000000000 +0100
@@ -86,7 +86,7 @@
   -version             print version information
 
 By default, 'file' is written in the MIFF image format.  To specify a particular image format, precede the filename with an image format name and a colon (i.e. ps:image) or specify the image type as the filename suffix (i.e. image.ps).  Specify 'file' as '-' for standard input or output.
-.SH SEE-ALSO
+.SH SEE ALSO
 ImageMagick(1)
 
 .SH COPYRIGHT
Index: imagemagick-6.3.7.9.dfsg1/utilities/mogrify.1.in
===================================================================
--- imagemagick-6.3.7.9.dfsg1.orig/utilities/mogrify.1.in	2007-12-26 04:10:09.000000000 +0100
+++ imagemagick-6.3.7.9.dfsg1/utilities/mogrify.1.in	2008-01-10 20:45:49.000000000 +0100
@@ -217,7 +217,7 @@
   -version             print version information
 
 By default, the image format of `file' is determined by its magic number.  To specify a particular image format, precede the filename with an image format name and a colon (i.e. ps:image) or specify the image type as the filename suffix (i.e. image.ps).  Specify 'file' as '-' for standard input or output.
-.SH SEE-ALSO
+.SH SEE ALSO
 ImageMagick(1)
 
 .SH COPYRIGHT
Index: imagemagick-6.3.7.9.dfsg1/utilities/montage.1.in
===================================================================
--- imagemagick-6.3.7.9.dfsg1.orig/utilities/montage.1.in	2007-12-26 04:10:09.000000000 +0100
+++ imagemagick-6.3.7.9.dfsg1/utilities/montage.1.in	2008-01-10 20:45:49.000000000 +0100
@@ -112,7 +112,7 @@
 In addition to those listed above, you can specify these standard X resources as command line options:  -background, -bordercolor, -borderwidth, -font, -mattecolor, or -title.
 
 By default, the image format of `file' is determined by its magic number.  To specify a particular image format, precede the filename with an image format name and a colon (i.e. ps:image) or specify the image type as the filename suffix (i.e. image.ps).  Specify 'file' as '-' for standard input or output.
-.SH SEE-ALSO
+.SH SEE ALSO
 ImageMagick(1)
 
 .SH COPYRIGHT
Index: imagemagick-6.3.7.9.dfsg1/utilities/stream.1.in
===================================================================
--- imagemagick-6.3.7.9.dfsg1.orig/utilities/stream.1.in	2007-12-26 04:10:09.000000000 +0100
+++ imagemagick-6.3.7.9.dfsg1/utilities/stream.1.in	2008-01-10 20:45:49.000000000 +0100
@@ -48,7 +48,7 @@
   -version             print version information
 
 By default, the image format of `file' is determined by its magic number.  To specify a particular image format, precede the filename with an image format name and a colon (i.e. ps:image) or specify the image type as the filename suffix (i.e. image.ps).  Specify 'file' as '-' for standard input or output.
-.SH SEE-ALSO
+.SH SEE ALSO
 ImageMagick(1)
 
 .SH COPYRIGHT
