Index: src/com/fluendo/player/Cortado.java
===================================================================
--- src/com/fluendo/player/Cortado.java	(revision 5453)
+++ src/com/fluendo/player/Cortado.java	(working copy)
@@ -30,8 +30,8 @@
         MouseListener, ComponentListener, BusHandler, StatusListener, ActionListener {
     private static final long serialVersionUID = 1L;
 
-    private static Cortado cortado;
-    private static CortadoPipeline pipeline;
+    private Cortado cortado;
+    private CortadoPipeline pipeline;
 
     private String urlString;
     private boolean audio;
@@ -192,10 +192,10 @@
       return res;
     }
 
-    public static void shutDown(Throwable error) {
+    public void shutDown(Throwable error) {
         Debug.log(Debug.INFO, "shutting down: reason: " + error.getMessage());
         error.printStackTrace();
-        cortado.stop();
+        stop();
     }
 
     public synchronized void init() {
@@ -341,7 +341,7 @@
         try {
             realRun();
         } catch (Throwable t) {
-            Cortado.shutDown(t);
+            shutDown(t);
         }
     }
 
