--- squid-2.6.PRE2/src/htcp.c.htcp	2006-06-06 06:37:41.000000000 +0000
+++ squid-2.6.PRE2/src/htcp.c	2006-06-12 15:09:52.000000000 +0000
@@ -1009,10 +1009,10 @@
     debug(31, 3) ("htcpHandleData: RR = %d\n", (int) hdr.RR);
     debug(31, 3) ("htcpHandleData: msg_id = %d\n", (int) hdr.msg_id);
     if (sz < hdr.length) {
-	debug(31, 0) ("htcpHandle: sz < hdr.length\n");
+	debug(31, 0) ("htcpHandleData: sz < hdr.length\n");
 	return;
     }
-    if (hdr.length + sizeof(htcpDataHeader) > sz) {
+    if (hdr.length + sizeof(htcpHeader) > sz) {
 	debug(31, 1) ("htcpHandleData: Invalid HTCP packet from %s\n", inet_ntoa(from->sin_addr));
 	return;
     }
