--- squid-2.6.PRE2/icons/Makefile.in.orig	2006-06-12 12:37:14.000000000 +0000
+++ squid-2.6.PRE2/icons/Makefile.in	2006-06-12 12:38:17.000000000 +0000
@@ -253,7 +253,7 @@
 		anthony-xbm.gif \
 		anthony-xpm.gif
 
-icondir = $(datadir)/icons
+icondir = $(pkgdatadir)/icons
 icon_DATA = $(ICON1) $(ICON2)
 EXTRA_DIST = $(ICON1) $(ICON2) icons.shar
 DISTCLEANFILES = 
--- squid-2.6.PRE2/errors/Makefile.in.orig	2006-06-12 12:46:24.000000000 +0000
+++ squid-2.6.PRE2/errors/Makefile.in	2006-06-12 12:48:10.000000000 +0000
@@ -216,8 +216,8 @@
 sharedstatedir = @sharedstatedir@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
-errordir = $(datadir)/errors
-DEFAULT_ERROR_DIR = $(errordir)
+errordir = $(pkgdatadir)/errors
+DEFAULT_ERROR_DIR = $(sysconfdir)/errors
 INSTALL_LANGUAGES = @ERR_LANGUAGES@
 LANGUAGES = \
 		Azerbaijani \
@@ -408,17 +408,17 @@
 
 install-data-local:
 	@for l in $(INSTALL_LANGUAGES); do \
-		$(mkinstalldirs) $(DESTDIR)$(DEFAULT_ERROR_DIR)/$$l && \
+		$(mkinstalldirs) $(DESTDIR)$(errordir)/$$l && \
 		for f in $(srcdir)/$$l/ERR_*; do \
-			echo "$(INSTALL_DATA) $$f $(DESTDIR)$(DEFAULT_ERROR_DIR)/$$l"; \
-			$(INSTALL_DATA) $$f $(DESTDIR)$(DEFAULT_ERROR_DIR)/$$l; \
+			echo "$(INSTALL_DATA) $$f $(DESTDIR)$(errordir)/$$l"; \
+			$(INSTALL_DATA) $$f $(DESTDIR)$(errordir)/$$l; \
 		done \
 	done
 
 uninstall-local:
 	@for l in $(INSTALL_LANGUAGES); do \
 		for f in $(srcdir)/$$l/ERR_*; do \
-	        	rm -f $(DESTDIR)$(DEFAULT_ERROR_DIR)/$$l/`basename $$f`; \
+	        	rm -f $(DESTDIR)$(errordir)/$$l/`basename $$f`; \
 		done \
 	done
 
@@ -431,13 +431,13 @@
 #
 addlang: all
 	-@if test -d $(srcdir)/$(ADDLANG); then \
-	if test -d $(DEFAULT_ERROR_DIR)/$(DESTLANG); then \
+	if test -d $(errordir)/$(DESTLANG); then \
 	cd $(srcdir)/$(ADDLANG)/ ; \
 	for f in ERR_*; do \
-		if test -f $(DEFAULT_ERROR_DIR)/$(DESTLANG)/$$f ; then \
+		if test -f $(errordir)/$(DESTLANG)/$$f ; then \
 			echo "appending $(ADDLANG)/$$f"; \
-			echo "<HR>" >> $(DEFAULT_ERROR_DIR)/$(DESTLANG)/$$f ; \
-			cat $$f >> $(DEFAULT_ERROR_DIR)/$(DESTLANG)/$$f; \
+			echo "<HR>" >> $(errordir)/$(DESTLANG)/$$f ; \
+			cat $$f >> $(errordir)/$(DESTLANG)/$$f; \
 		fi; \
 	done; \
 	fi \
--- squid-2.6.STABLE2/src/Makefile.in.build	2006-08-02 02:32:53.000000000 +0000
+++ squid-2.6.STABLE2/src/Makefile.in	2006-08-02 02:53:16.000000000 +0000
@@ -588,18 +588,18 @@
 DEFAULT_CONFIG_FILE = $(sysconfdir)/squid.conf
 DEFAULT_MIME_TABLE = $(sysconfdir)/mime.conf
 DEFAULT_DNSSERVER = $(libexecdir)/`echo dnsserver | sed '$(transform);s/$$/$(EXEEXT)/'`
-DEFAULT_LOG_PREFIX = $(localstatedir)/logs
+DEFAULT_LOG_PREFIX = $(localstatedir)/log/squid
 DEFAULT_CACHE_LOG = $(DEFAULT_LOG_PREFIX)/cache.log
 DEFAULT_ACCESS_LOG = $(DEFAULT_LOG_PREFIX)/access.log
 DEFAULT_STORE_LOG = $(DEFAULT_LOG_PREFIX)/store.log
-DEFAULT_PID_FILE = $(DEFAULT_LOG_PREFIX)/squid.pid
-DEFAULT_SWAP_DIR = $(localstatedir)/cache
+DEFAULT_PID_FILE = $(localstatedir)/run/squid.pid
+DEFAULT_SWAP_DIR = $(localstatedir)/spool/squid
 DEFAULT_PINGER = $(libexecdir)/`echo pinger | sed '$(transform);s/$$/$(EXEEXT)/'`
 DEFAULT_UNLINKD = $(libexecdir)/`echo unlinkd | sed '$(transform);s/$$/$(EXEEXT)/'`
 DEFAULT_DISKD = $(libexecdir)/`echo diskd-daemon | sed '$(transform);s/$$/$(EXEEXT)/'`
-DEFAULT_ICON_DIR = $(datadir)/icons
-DEFAULT_ERROR_DIR = $(datadir)/errors/@ERR_DEFAULT_LANGUAGE@
-DEFAULT_MIB_PATH = $(datadir)/mib.txt
+DEFAULT_ICON_DIR = $(pkgdatadir)/icons
+DEFAULT_ERROR_DIR = $(pkgdatadir)/errors/@ERR_DEFAULT_LANGUAGE@
+DEFAULT_MIB_PATH = $(sysconfdir)/mib.txt
 DEFAULT_HOSTS = @OPT_DEFAULT_HOSTS@
 
 # Don't automatically uninstall config files
@@ -844,20 +844,20 @@
 uninstall-info-am:
 install-dataDATA: $(data_DATA)
 	@$(NORMAL_INSTALL)
-	test -z "$(datadir)" || $(mkdir_p) "$(DESTDIR)$(datadir)"
+	test -z "$(datadir)" || $(mkdir_p) "$(DESTDIR)$(sysconfdir)/squid"
 	@list='$(data_DATA)'; for p in $$list; do \
 	  if test -f "$$p"; then d=; else d="$(srcdir)/"; fi; \
 	  f=$(am__strip_dir) \
-	  echo " $(dataDATA_INSTALL) '$$d$$p' '$(DESTDIR)$(datadir)/$$f'"; \
-	  $(dataDATA_INSTALL) "$$d$$p" "$(DESTDIR)$(datadir)/$$f"; \
+	  echo " $(dataDATA_INSTALL) '$$d$$p' '$(DESTDIR)$(sysconfdir)/$$f'"; \
+	  $(dataDATA_INSTALL) "$$d$$p" "$(DESTDIR)$(sysconfdir)/$$f"; \
 	done
 
 uninstall-dataDATA:
 	@$(NORMAL_UNINSTALL)
 	@list='$(data_DATA)'; for p in $$list; do \
 	  f=$(am__strip_dir) \
-	  echo " rm -f '$(DESTDIR)$(datadir)/$$f'"; \
-	  rm -f "$(DESTDIR)$(datadir)/$$f"; \
+	  echo " rm -f '$(DESTDIR)$(sysconfdir)/$$f'"; \
+	  rm -f "$(DESTDIR)$(sysconfdir)/$$f"; \
 	done
 install-sysconfDATA: $(sysconf_DATA)
 	@$(NORMAL_INSTALL)
