diff -uNrp php5-5.2.1.orig/ext/session/session.c php5-5.2.1/ext/session/session.c
--- php5-5.2.1.orig/ext/session/session.c	2007-01-09 23:04:49.000000000 -0800
+++ php5-5.2.1/ext/session/session.c	2007-04-23 15:30:26.000000000 -0700
@@ -848,6 +848,7 @@ new_session:
 	} else if (PS(invalid_session_id)) { /* address instances where the session read fails due to an invalid id */
 		PS(invalid_session_id) = 0;
 		efree(PS(id));
+		PS(id) = NULL;
 		goto new_session;
 	}
 }
@@ -1577,6 +1578,7 @@ PHP_FUNCTION(session_regenerate_id)
 				RETURN_FALSE;
 			}
 			efree(PS(id));
+			PS(id) = NULL;
 		}
 	
 		PS(id) = PS(mod)->s_create_sid(&PS(mod_data), NULL TSRMLS_CC);
