diff -uNrp php5-5.2.1.orig/ext/standard/string.c php5-5.2.1/ext/standard/string.c
--- php5-5.2.1.orig/ext/standard/string.c	2007-02-01 05:45:25.000000000 -0800
+++ php5-5.2.1/ext/standard/string.c	2007-04-23 15:25:37.000000000 -0700
@@ -4631,18 +4631,20 @@ PHP_FUNCTION(substr_count)
 			php_error_docref(NULL TSRMLS_CC, E_WARNING, "Offset should be greater than or equal to 0.");
 			RETURN_FALSE;		
 		}
-		p += Z_LVAL_PP(offset);
-		if (p > endp) {
+
+		if (Z_LVAL_PP(offset) > Z_STRLEN_PP(haystack)) {
 			php_error_docref(NULL TSRMLS_CC, E_WARNING, "Offset value %ld exceeds string length.", Z_LVAL_PP(offset));
 			RETURN_FALSE;		
 		}
+		p += Z_LVAL_PP(offset);
+
 		if (ac == 4) {
 			convert_to_long_ex(length);
 			if (Z_LVAL_PP(length) <= 0) {
 				php_error_docref(NULL TSRMLS_CC, E_WARNING, "Length should be greater than 0.");
 				RETURN_FALSE;		
 			}
-			if ((p + Z_LVAL_PP(length)) > endp) {
+			if (Z_LVAL_PP(length) > (Z_STRLEN_PP(haystack) - Z_LVAL_PP(offset))) {
 				php_error_docref(NULL TSRMLS_CC, E_WARNING, "Length value %ld exceeds string length.", Z_LVAL_PP(length));
 				RETURN_FALSE;
 			}
@@ -5063,11 +5065,16 @@ PHP_FUNCTION(substr_compare)
 		offset = (offset < 0) ? 0 : offset;
 	}
 
-	if ((offset + len) > s1_len) {
+	if(offset > s1_len) {
 		php_error_docref(NULL TSRMLS_CC, E_WARNING, "The start position cannot exceed initial string length");
 		RETURN_FALSE;
 	}
 
+	if(len > s1_len - offset) {
+		php_error_docref(NULL TSRMLS_CC, E_WARNING, "The length cannot exceed initial string length");
+		RETURN_FALSE;
+	}
+
 	cmp_len = (uint) (len ? len : MAX(s2_len, (s1_len - offset)));
 
 	if (!cs) {
diff -uNrp php5-5.1.6-unpatched/ext/standard/tests/strings/substr_compare.phpt php5-5.1.6/ext/standard/string.c
--- php5-5.1.6-unpatched/ext/standard/tests/strings/substr_compare.phpt	2007-04-23 16:12:26.000000000 -0700
+++ php5-5.1.6/ext/standard/tests/strings/substr_compare.phpt	2007-04-23 16:16:15.000000000 -0700
@@ -24,7 +24,7 @@
 int(1)
 int(-1)
 
-Warning: substr_compare(): The start position cannot exceed initial string length in %s on line %d
+Warning: substr_compare(): The length cannot exceed initial string length in %s on line %d
 bool(false)
 
 Warning: substr_compare() expects parameter 5 to be boolean, object given in %s on line %d
