diff -uNrp php5-5.2.1.orig/ext/session/session.c php5-5.2.1/ext/session/session.c
--- php5-5.2.1.orig/ext/session/session.c	2007-01-09 23:04:49.000000000 -0800
+++ php5-5.2.1/ext/session/session.c	2007-04-23 15:21:25.000000000 -0700
@@ -478,7 +478,7 @@ PS_SERIALIZER_DECODE_FUNC(php_binary)
 		zval **tmp;
 		namelen = *p & (~PS_BIN_UNDEF);
 
-		if (namelen > PS_BIN_MAX || (p + namelen) >= endptr) {
+		if (namelen < 0 || namelen > PS_BIN_MAX || (p + namelen) >= endptr) {
 			return FAILURE;
 		}
 
