diff -uNrp php5-5.2.1.orig/ext/standard/var_unserializer.re php5-5.2.1/ext/standard/var_unserializer.re
--- php5-5.2.1.orig/ext/standard/var_unserializer.re	2006-12-14 16:58:08.000000000 -0800
+++ php5-5.2.1/ext/standard/var_unserializer.re	2007-04-23 15:35:45.000000000 -0700
@@ -138,12 +138,18 @@ PHPAPI void var_destroy(php_unserialize_
 
 /* }}} */
 
-static char *unserialize_str(const unsigned char **p, int len)
+static char *unserialize_str(const unsigned char **p, size_t *len)
 {
-	int i, j;
-	char *str = emalloc(len+1);
+	size_t i, j;
+	char *str = safe_emalloc(*len, 1, 1);
+	unsigned char *end = *p+*len;
 
-	for (i = 0; i < len; i++) {
+	if(end < *p) {
+		efree(str);
+		return NULL;
+	}
+
+	for (i = 0; i < *len && *p < end; i++) {
 		if (**p != '\\') {
 			str[i] = (char)**p;
 		} else {
@@ -167,6 +173,7 @@ static char *unserialize_str(const unsig
 		(*p)++;
 	}
 	str[i] = 0;
+	*len = i;
 	return str;
 }
 
@@ -518,7 +525,7 @@ PHPAPI int php_var_unserialize(UNSERIALI
 		return 0;
 	}
 
-	if ((str = unserialize_str(&YYCURSOR, len)) == NULL) {
+	if ((str = unserialize_str(&YYCURSOR, &len)) == NULL) {
 		return 0;
 	}
 
