diff -uNrp php5-5.2.1.orig/ext/mbstring/mbstring.c php5-5.2.1/ext/mbstring/mbstring.c
--- php5-5.2.1.orig/ext/mbstring/mbstring.c	2007-01-12 04:06:45.000000000 -0800
+++ php5-5.2.1/ext/mbstring/mbstring.c	2007-04-23 15:43:18.000000000 -0700
@@ -3301,8 +3301,8 @@ PHP_FUNCTION(mb_decode_numericentity)
 
 #define SKIP_LONG_HEADER_SEP_MBSTRING(str, pos)										\
 	if (str[pos] == '\r' && str[pos + 1] == '\n' && (str[pos + 2] == ' ' || str[pos + 2] == '\t')) {	\
-		pos += 3;											\
-		while (str[pos] == ' ' || str[pos] == '\t') {							\
+		pos += 2;											\
+		while (str[pos + 1] == ' ' || str[pos + 1] == '\t') {							\
 			pos++;											\
 		}												\
 		continue;											\
diff -uNrp php5-5.2.1.orig/ext/standard/mail.c php5-5.2.1/ext/standard/mail.c
--- php5-5.2.1.orig/ext/standard/mail.c	2007-01-24 16:26:51.000000000 -0800
+++ php5-5.2.1/ext/standard/mail.c	2007-04-23 15:43:18.000000000 -0700
@@ -48,8 +48,8 @@
 
 #define SKIP_LONG_HEADER_SEP(str, pos)										\
 	if (str[pos] == '\r' && str[pos + 1] == '\n' && (str[pos + 2] == ' ' || str[pos + 2] == '\t')) {	\
-		pos += 3;											\
-		while (str[pos] == ' ' || str[pos] == '\t') {							\
+		pos += 2;											\
+		while (str[pos + 1] == ' ' || str[pos + 1] == '\t') {							\
 			pos++;											\
 		}												\
 		continue;											\
